app/controllers/claims_controller.rb

```diff
@@ -23,10 +23,13 @@ def show
     if params[:slug] == "postcode-search" && postcode
       redirect_to claim_path(current_policy_routing_name, "select-home-address", {"claim[postcode]": params[:claim][:postcode], "claim[address_line_1]": params[:claim][:address_line_1]}) and return unless invalid_postcode?
     elsif params[:slug] == "select-home-address" && postcode
+      session[:claim_postcode] = params[:claim][:postcode]
+      session[:claim_address_line_1] = params[:claim][:address_line_1]
       if address_data.nil?
```

Comment:
Shouldn't we clear the postcode and address from the session here?

Reply:
Ideally yes, but if people press back those pages pick up values from the session, this completely needs rework if we want to ditch the session issues with pressing the back button. This keeps the session in sync and avoids that issue you mentioned about having a stale value when you press back.